### PR TITLE
dissect: don't declare unused variables on archs that have no GPT dis…

### PR DESCRIPTION
**Pull request by poettering**
…covery

Fixes: #17839

### CODE REVIEW

File: src/shared/dissect-image.c [hunk 1]
```diff
@@ -460,8 +460,12 @@ int dissect_image(
                 DissectedImage **ret) {
 
 #if HAVE_BLKID
-        sd_id128_t root_uuid = SD_ID128_NULL, root_verity_uuid = SD_ID128_NULL,
-                usr_uuid = SD_ID128_NULL, usr_verity_uuid = SD_ID128_NULL;
+#ifdef GPT_ROOT_NATIVE
```
Comment (bluca) [inline src/shared/dissect-image.c:463]: The variables are referenced slightly below, lines 511-518 - so there are more ifdefs needed

Comment (poettering) [reply src/shared/dissect-image.c:463]: indeed!

**Comment by poettering** [comment]
@bluca btw, i added you to the systemd org on github now, so that you can label things and merge stuff. Please use your new powers wisely, and if in doubt, ask. Our general rule for merging: one review required. Welcome!

**Comment by poettering** [comment]
force pushed a new version with a bit more ifdeffery, as requested. ptal!

**Review by bluca — APPROVED** [review]
Looks good, also did a few compilation tests on my machine

**Comment by bluca** [comment]
> @bluca btw, i added you to the systemd org on github now, so that you can label things and merge stuff. Please use your new powers wisely, and if in doubt, ask. Our general rule for merging: one review required. Welcome!

Oh wow, thank you! I will be careful, and I'll help as best as I can!

**Pull request merged**